### PR TITLE
mouse: Increase number of buttons supported from 5 to 7

### PR DESCRIPTION
**Pull request by cjmayo**
Full support for mouse with wheel and thumb buttons

N.B. 5 buttons was still hardcoded in ButtonTranslator.cpp

### CODE REVIEW

**Comment by cjmayo** [comment]
Also needed by 5 button Logitech Wireless Trackball M570 which sends codes 8 & 9 for thumb buttons.

File: xbmc/input/MouseStat.h [hunk 1]
```diff
@@ -32,20 +32,24 @@
...
 #define MOUSE_MINIMUM_MOVEMENT 2
 #define MOUSE_DOUBLE_CLICK_LENGTH 500L
 #define MOUSE_ACTIVE_LENGTH   5000L
 
-#define MOUSE_MAX_BUTTON 5
+#define MOUSE_MAX_BUTTON 7
 
 enum MOUSE_STATE { MOUSE_STATE_NORMAL = 1, MOUSE_STATE_FOCUS, MOUSE_STATE_DRAG, MOUSE_STATE_CLICK };
-enum MOUSE_BUTTON { MOUSE_LEFT_BUTTON = 0, MOUSE_RIGHT_BUTTON, MOUSE_MIDDLE_BUTTON, MOUSE_EXTRA_BUTTON1, MOUSE_EXTRA_BUTTON2 };
+enum MOUSE_BUTTON { MOUSE_LEFT_BUTTON = 0, MOUSE_RIGHT_BUTTON, MOUSE_MIDDLE_BUTTON, MOUSE_EXTRA_BUTTON1, MOUSE_EXTRA_BUTTON2, MOUSE_EXTRA_BUTTON3, MOUSE_EXTRA_BUTTON4 };
```
Comment (mkortstiege) [inline xbmc/input/MouseStat.h]: Please re-align the MOUSE_BUTTON enum (like MOUSE_STATE). That line is way too long ;)

**Comment by mkortstiege** [comment]
I think you need a rebase here. This is not based on a recent checkout as it seems.

**Comment by cjmayo** [comment]
Rebased and reformatted enum MOUSE_BUTTON to match latest layout of MOUSE_STATE.

**Comment by mkortstiege** [comment]
Thanks. Looks good to me. jenkins build this please.

**Comment by mkortstiege** [comment]
@Paxxi objections to shove this in?

**Comment by Paxxi** [comment]
shove it in, looks good

**Pull request merged**